hwdetect/hwdetect.sh

```diff
@@ -21,8 +21,11 @@ function find_usb_ttl()
 
 function find_sd_card()
 {
-	ls /dev | grep mmcblk
-	lsusb | grep -i storage
+	(ls /dev | grep mmcblk) > newsdcard
+	lsusb | grep -i storage >> newsdcard
+	diff sdcard newsdcard | awk -F'>' '{if ($2!="") print "Added ", $2}'
+	diff sdcard newsdcard | awk -F'<' '{if ($2!="") print "Removed ", $2}'
+	cat newsdcard > sdcard
```

Comment:
Creation files in working directory when user doesn't expect this is definitely bad idea.
You should use `/tmp/`, etc. for such purpose.
Or if you don't need to preserve state between launches - files aren't needed at all - variables should be enough.

BTW, have you seen `lsusb` showing device with "storage" in description?
Is it block device?
(Because in my case only cardreader itself is seen and it doesn't have "storage" in description)

Reply:
I have no SDcard reader in GL laptop, but I have separate SDCard reader via usb and it is detected as usb staorage device. It is observed as 'sda' but not 'mmcblk'